build.assets/tooling/cmd/notarize-apple-binaries/gonConfig.go

```diff
@@ -0,0 +1,170 @@
+	flag.StringVar(&gc.ApplePassword, "apple-password", "", "Apple Connect password used for notarization")
+
+	return gc
+}
+
+func (gc *GonConfig) Check() error {
+	err := gc.validateAppleUsername()
+	if err != nil {
+		return trace.Wrap(err, "failed to validate the apple-username flag")
+	}
+
+	err = gc.validateApplePassword()
+	if err != nil {
+		return trace.Wrap(err, "failed to validate the apple-password flag")
+	}
+
+	err = gc.validateBinaryPaths()
+	if err != nil {
+		return trace.Wrap(err, "failed to validate binary path(s)")
+	}
+
+	return nil
+
+	// It might be worth adding an actual login check here in the future
+}
+
+func (gc *GonConfig) validateAppleUsername() error {
+	if gc.AppleUsername == "" {
+		return trace.BadParameter("the apple-username flag should not be empty")
+	}
+
+	return nil
+}
+
+func (gc *GonConfig) validateApplePassword() error {
+	if gc.ApplePassword == "" {
+		return trace.BadParameter("the apple-password flag should not be empty")
+	}
+
+	return nil
+}
+
+func (gc *GonConfig) validateBinaryPaths() error {
+	// Check for minimum arg count and load the parameters into the struct
+	if flag.NArg() == 0 {
+		return trace.BadParameter("the path to at least one binary is required")
+	}
+	gc.BinaryPaths = flag.Args()
+
+	// Validate each path
+	binaryPaths := flag.Args()
+	for _, binaryPath := range binaryPaths {
+		logrus.Debugf("Validating binary path %q...", binaryPath)
+		err := gc.verifyFileIsValidForSigning(binaryPath)
+		if err != nil {
+			return trace.Wrap(err, "file %q failed binary validation", binaryPath)
+		}
+	}
+
+	return nil
+}
+
+// Returns an error if the file is not valid for signing
+func (gc *GonConfig) verifyFileIsValidForSigning(filePath string) error {
+	err := gc.verifyPathIsNotDirectory(filePath)
+	if err != nil {
+		return trace.Wrap(err, "filesystem path %q is a directory")
+	}
+
+	err = gc.verifyFileIsAppleBinary(filePath)
+	if err != nil {
+		return trace.Wrap(err, "file %q is not an Apple binary", filePath)
+	}
+
+	return nil
+}
+
+// Returns an error if the provided binary path is actually a directory
+func (gc *GonConfig) verifyPathIsNotDirectory(filePath string) error {
+	fileInfo, err := os.Stat(filePath)
+	if err != nil {
+		return trace.Wrap(err, "failed to stat file system information for path %q", filePath)
+	}
+
+	if fileInfo.IsDir() {
+		return trace.Errorf("filesystem path %q is a directory and does not point to a binary", filePath)
+	}
+
+	return nil
+}
+
+// Returns an error if the file is not a valid Apple binary
+// Effectively does `file $BINARY | grep -ic 'mach-o'`
+func (gc *GonConfig) verifyFileIsAppleBinary(filePath string) error {
+	// Note that this may also match Java bytecode but writing around
+	// that corner case is not worth the time required.
+	// See https://github.com/file/file/blob/master/magic/Magdir/mach for details
+	fileHandle, err := os.Open(filePath)
+	if err != nil {
+		return trace.Wrap(err, "failed to read file %q for reading", filePath)
+	}
+
```

Comment:
Don't forget to `defer fileHandle.Close()`

Reply:
Thanks, missed that - added